extensions/elasticsearch-rest-client-common/deployment/src/main/java/io/quarkus/elasticsearch/restclient/common/deployment/DevServicesElasticsearchProcessor.java

```diff
@@ -209,11 +210,15 @@ private DevServicesResultBuildItem.RunningDevService startElasticsearchDevServic
             container.withReuse(config.reuse);
 
             container.start();
+
+            var httpHost = container.getHost() + ":" + container.getMappedPort(ELASTICSEARCH_PORT);
+            if (createdContainer.hostName() != null) {
+                httpHost = createdContainer.hostName() + ":" + ELASTICSEARCH_PORT;
```

Comment:
On second thought... Shouldn't it be this?

```suggestion
                httpHost = createdContainer.hostName() + ":" + container.getMappedPort(ELASTICSEARCH_PORT);
```

Reply:
Container's port within shared network is accessible via direct port without mapping.

Similar code occurs in properly working `extensions/devservices/postgresql/src/main/java/io/quarkus/devservices/postgresql/deployment/PostgresqlDevServicesProcessor.java`. There we can find similar code with fixed port:
```java
        public String getEffectiveJdbcUrl() {
            if (useSharedNetwork) {
                String additionalUrlParams = constructUrlParameters("?", "&");
                return "jdbc:postgresql://" + hostName + ":" + POSTGRESQL_PORT
                        + "/" + getDatabaseName() + additionalUrlParams;
            } else {
                return super.getJdbcUrl();
            }
        }
```

Reply:
Makes sense, thanks.